commons/zenoh-protocol/src/core/locator.rs

```diff
@@ -37,6 +37,16 @@ impl Locator {
         Ok(Self(ep))
     }
 
+    pub fn new_with_config<A, B, C>(protocol: A, address: B, metadata: C, config: String) -> ZResult<Self>
```

Comment:
A `Locator` by definition has no `Config` while an `EndPoint` has.
`Metadata` is meant to be transmitted over the network to signal the receiver additional information. 
So, this addition is not only semantically incorrect but also wrong since `EndPoint -> Locator` conversion will trim out the `Config` part.

Reply:
Introduced a HelloEndPoint struct. Please have a look